src/arch/cortex-m0/syscalls.S

```diff
@@ -0,0 +1,34 @@
+.cpu cortex-m0
+.syntax unified
+.thumb
+.text
+
+.section .syscalls
+
+/* Cortex-M0 can only push/pop registers R0-R7 directly, so move R8-R11 to/from
+ * R0-R3. This is equivalent to the 32-bit "push/pop {r4-r11}" instructions. */
+
+.macro SYSCALL NAME, NUM
+.global \NAME
+.thumb_func
+\NAME :
```

Comment:
I like this trick. Perhaps it would make sense to have each architecture define just the macro, and then call the macro (as you do below in lines 31-34) so there is no chance of getting the syscall numbers inconsistent.